Symfony/CS/Tokenizer/Tokens.php

```diff
@@ -751,6 +751,94 @@ public function getPrevMeaningfulToken($index)
     }
 
     /**
+     * Find a sequence of meaningful tokens and returns the array of their locations.
+     *
+     * @param array      $sequence      an array of tokens (same format used by getNextTokenOfKind)
+     * @param int        $start         start index, defaulting to the start of the file
+     * @param int        $end           end index, defaulting to the end of the file
+     * @param bool|array $caseSensitive global case sensitiveness or an array of booleans, whose keys should match
+     *                                  the ones used in $others. If any is missing, the default case-sensitive
+     *                                  comparison is used.
+     *
+     * @return array|null an array containing the tokens matching the sequence elements, indexed by their position
+     */
+    public function findSequence(array $sequence, $start = 0, $end = null, $caseSensitive = true)
+    {
+        // $end defaults to the end of the file
```

Comment:
We're iterating a collection, not a file.
We may build `Tokens` collection without file.